spring-messaging/src/main/java/org/springframework/messaging/rsocket/annotation/support/RSocketPayloadReturnValueHandler.java

```diff
@@ -50,22 +56,56 @@ public class RSocketPayloadReturnValueHandler extends AbstractEncoderMethodRetur
 	 */
 	public static final String RESPONSE_HEADER = "rsocketResponse";
 
+	private RSocketStrategies strategies;
+	private MimeType compositeMime =  MimeTypeUtils.parseMimeType(WellKnownMimeType.MESSAGE_RSOCKET_COMPOSITE_METADATA.getString());
 
 	public RSocketPayloadReturnValueHandler(List<Encoder<?>> encoders, ReactiveAdapterRegistry registry) {
 		super(encoders, registry);
 	}
+	public RSocketPayloadReturnValueHandler(RSocketStrategies strategies, ReactiveAdapterRegistry registry) {
+		this(strategies.encoders(), registry);
+		this.strategies = strategies;
+	}
 
 
 	@Override
 	protected Mono<Void> handleEncodedContent(
 			Flux<DataBuffer> encodedContent, MethodParameter returnType, Message<?> message) {
 
 		AtomicReference<Flux<Payload>> responseRef = getResponseReference(message);
+		AtomicReference<MimeType> contentTypeAtomic = getResponseContentType(message);
 		Assert.notNull(responseRef, "Missing '" + RESPONSE_HEADER + "'");
-		responseRef.set(encodedContent.map(PayloadUtils::createPayload));
+		responseRef.set(encodedContent.switchOnFirst((signal, inner) -> {
```

Comment:
We shouldn't automatically use `switchOnFirst`. If no content negotiation took place, and the application is using the connection-wide `dataMimeType`, there is no need to use this operator.